packages/liquidation-sdk-viem/src/preLiquidation/positionGetters.ts

```
@@ -4,69 +4,99 @@ import { fetchAccrualPosition } from "@morpho-org/blue-sdk-viem";
+          const collateralAsset =
+            market !== null ? market[0]?.collateralAsset : undefined;
 
-  const preLiquidablePositions = await Promise.all(
-    preLiquidationInstances
-      .filter((position) => position !== undefined)
-      .map(async (preLiquidationPosition) => {
-        return await Promise.all(
-          preLiquidationPosition.borrowers.map(async (borrower) => {
-            return await getPreLiquidablePosition(
-              client,
-              preLiquidationPosition.preLiquidation,
-              borrower,
-              preLiquidationPosition.collateralAsset,
-              preLiquidationPosition.loanAsset,
-            );
-          }),
-        );
-      }),
-  );
+          if (
+            loanAsset === undefined ||
+            collateralAsset === undefined ||
+            collateralAsset === null
+          )
+            return;
+
+          return {
+            ...preLiquidation,
+            loanAsset,
+            collateralAsset,
+          };
+        }),
+    );
+
+    const preLiquidablePositions = await Promise.all(
+      preLiquidationInstances
+        .filter((position) => position !== undefined)
+        .map(async (preLiquidationPosition) => {
+          return await Promise.all(
+            preLiquidationPosition.enabledPositions.map(async (borrower) => {
+              return await getPreLiquidablePosition(
+                client,
+                preLiquidationPosition,
+                borrower,
+                preLiquidationPosition.collateralAsset,
+                preLiquidationPosition.loanAsset,
+              );
+            }),
+          );
+        }),
+    );
 
-  return preLiquidablePositions
-    .flat()
-    .filter((position) => position.preSeizableCollateral !== undefined);
+    return preLiquidablePositions
+      .flat()
+      .filter((position) => position.preSeizableCollateral !== undefined);
+  } catch (error) {
```

There are many reasons why this execution would throw (RPC error, indexer error, API error, ...)
All of these paths are handled the same way: log the error to the console and return an empty array

This is an anti-pattern for the consumer: they cannot handle the different errors however they want

Instead, just let the function fail and let the consumer handle the errors (ideally document them in the jsdoc!)